fusesoc/build/vivado.py

```diff
@@ -0,0 +1,202 @@
+import os.path
+from fusesoc import utils
+
+from fusesoc.build.backend import Backend
+
+""" Vivado Backend
+
+The Vivado backend executes Xilinx Vivado to build systems and program the FPGA.
+
+The backend defines the following section:
+
+    [vivado]
+    part = <part name> # Format <family><device><package>-<speedgrade>
+    hw_device = <device name> # Format <family><device>_0
+
+A core (usually the system core) can add the following files:
+
+- Standard design sources
+
+- Constraints: Supply xdc files with file_type=xdc
+
+- IP: Supply the IP core xdi file with file_type=xdi and other files (like .prj)
+      as file_type=data
+"""
+class Vivado(Backend):
+    """ Define the toolname. It is used by FuseSoC to select the backend"""
+    TOOL_NAME = 'vivado'
+
+    """ The constructor of the backend"""
+    def __init__(self, system):
+        # Do the general backend initialization
+        super(Vivado, self).__init__(system)
+        # Set work root. That is the name of the output path for build files
+        self.work_root = os.path.join(self.build_root, 'bld-{}'.format(self.TOOL_NAME))
+
+    """ Configuration is the first phase of the build
+
+    In the vivado backend the project TCL is written and all files are copied
+    """
+    def configure(self, args):
+        super(Vivado, self).configure(args)
+
+        self._write_project_tcl_file()
+
+    """ Write the project tcl file
+
+    This writes the project tcl file. It first collects all sources, ip and contraints
+    and then writes them to the tcl file along with the build steps.
+    """
+    def _write_project_tcl_file(self):
+        # Get the synthesis files and files specific to vivado
+        (src, incdirs) = self._get_fileset_files(['vivado', 'synth'])
+
+        ip = []         # IP descriptions (xci files)
+        constr = []     # Constraints (xdc files)
+        verilog = []    # (System) Verilog files
+        vhdl = []       # VHDL files
+
+        hasvhdl2008 = False
+
+        for s in src:
+            if s.file_type == 'xci':
+                ip.append(s.name)
+            elif s.file_type == 'xdc':
+                constr.append(s.name)
+            elif s.file_type.startswith('verilogSource'):
+                verilog.append(s.name)
+            elif s.file_type.startswith('systemVerilogSource'):
+                verilog.append(s.name)
+            elif s.file_type.startswith('vhdlSource'):
+                vhdl.append(s.name)
+                if s.file_type == 'vhdlSource-2008':
```

Comment:
I think you still need to load vhdl 2008 files with `read_vhdl -vhdl2008 $file`, and you also need to know which library to put each of the VHDL files in. My suggestion would be to generate the complete tcl command in this loop instead of deferring it to line 93. Then you can do something like `tcl_file.write('read_vhdl -vhdl2008 -library {} {}'.format(s.logical_name, s.name)` for vhdl 2008 files, `tcl_file.write('read_verilog -sv {}'.format(s.name)` for SystemVerilog files etc.